crates/cext/src/sparse_observable.rs

```diff
@@ -718,6 +720,100 @@ pub unsafe extern "C" fn qk_obs_compose_map(
     Box::into_raw(Box::new(result))
 }
 
+/// @ingroup QkObs
+/// Apply a new qubit layout to the observable.
+///
+/// @param obs A pointer to the observable, this observable will be modified in place upon success.
+/// Check the exit code to ensure the layout was correctly applied.
+/// @param layout A pointer to the layout. The pointer must point to an array to
+/// ``qk_obs_num_qubits(obs)`` elements of type ``uint32_t``. Each element must have values
+/// in ``[0, num_qubits)``.
```

Comment:
What does the array passed to layout represent? I assume it's mapping index i to the physical qubit. But it's good to be explicit here about what the alayout array is besides just the data types.

Reply:
good point, I added an explanation to the docstring in cf65b46 👍🏻